x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/CrossClusterAccessSubjectInfo.java

```diff
@@ -69,6 +89,15 @@ public Authentication getAuthentication() {
         return authentication;
     }
 
+    public CrossClusterAccessSubjectInfo cleanWithValidation() {
+        // Need to do this first. Otherwise, the `copyWithFilteredMetadataFields` call in `copyAuthenticationWithCleanMetadata` may fail
+        // with a confusing error message for unsupported types
+        ensureAuthenticationHasSupportedSubjectType();
+        final var cleanCopy = new CrossClusterAccessSubjectInfo(copyAuthenticationWithCleanMetadata(), roleDescriptorsBytesList);
+        cleanCopy.validate();
+        return cleanCopy;
```

Comment:
Why cannot we call `validate()` first? Validate internally check type first and does not seem to rely on the clean copy. 
We also might not need assert the type in `copyAuthenticationWithCleanMetadata` since as is it is checked again in `validate`. If we move `validate` to be the first call. We can just assert the type here afer the cleanCopy if we really want to.

Reply:
Mostly because I want to make sure we validate the object we actually use in the remaining flow. It seems a little counter intuitive to validate, then change the object, then assume it's still valid. If our validation logic evolves to include more metadata checks we fail when we shouldn't (because we fail on something being present that would get cleaned) and not fail when we should (because we cleaned too much). Ideally, we wouldn't even check the type first but that leads to confusing error messages, since `copyAuthenticationWithCleanMetadata` doesn't always produce valid Authentication instance for unsupported types.

Reply:
LMK if you feel strongly, otherwise I'll keep (largely) as is. I've pushed a change to inlines the type check and asserts `!authentication.isCrossCluster()` in other places since I've adopted your deny-list suggestion.

Reply:
No strong feeling at all. Thanks for asking.